qlib/TableMapper.qm

```diff
@@ -663,13 +665,14 @@ on_error table_mapper.rollback();
             # copy all runtime mappings to the output hash
             map dh{$1.key} = m_runtime{$1.value}, rconsth.pairIterator();
 
-            int size = rec.firstValue().size();
-            map mapFieldIntern(\dh, $1, rec, size), mapd.keyIterator();
-            count += size;
+            map mapFieldIntern(\dh, $1, rec, True, rec_list_size), mapd.keyIterator();
+            count += rec_list_size;
 
             # map record data to get the keys for the buffer
             if (!hbuf) {
                 # copy all constant mappings to the output hash
+                # FIXME why copying const dat here again even if it is
```

Comment:
ok, I was misleaded by lines `if (exists m.constant)' and 'else if (exists m.runtime) {' in mapFieldIntern() that are actually also mentioned in your comments above - thanks for clarifying, I have to think about this one